### PR TITLE
Move Clippy to extra CI steps

### PR DESCRIPTION
**Pull request by domenukk**
This makes it possible to see if clippy updated or the PR is functionally broken

### CODE REVIEW

**Comment by domenukk** [comment]
Seems like an update broke CI already... Help appreciated, I don't have time to figure this one out atm

**Comment by domenukk** [comment]
This doesn't work, no clue why. Giving up..

**Comment by domenukk** [comment]
Nope :/

**Comment by domenukk** [comment]
No clue why this doesn't like me. Closing for good..

**Comment by domenukk** [comment]
This might again be feasible if #1768 is resolved.

**Comment by domenukk** [comment]
The error is a bunch of this:

```rust
error[E0407]: method `push_path_constraint` is not a member of trait `Runtime`
   --> libafl_concolic/symcc_runtime/src/tracing.rs:184:5
    |
184 | /     fn push_path_constraint(&mut self, constraint: RSymExpr, taken: bool, site_id: usize) {
185 | |         self.write_message(SymExpr::PathConstraint {
186 | |             constraint,
187 | |             taken,
188 | |             location: site_id.into(),
189 | |         });
190 | |     }
    | |_____^ not a member of trait `Runtime`
```

**Comment by tokatoka** [comment]
let's see if it's fixed

**Comment by tokatoka** [comment]
The issue is that bindgen does not generate bindings correctly with latest llvm

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -107,6 +108,28 @@ jobs:
...
+    - name: Remove Dotnet & Haskell
+      run: rm -rf /usr/share/dotnet && rm -rf /opt/ghc
+    - uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+    - name: Install and cache deps
+      uses: awalsh128/cache-apt-pkgs-action@v1.1.0
+      with:
+        packages: llvm llvm-dev clang ninja-build clang-format-13 shellcheck libgtk-3-dev gcc-arm-linux-gnueabi g++-arm-linux-gnueabi libslirp-dev libz3-dev
```
Comment (tokatoka) [inline .github/workflows/build_and_test.yml:124]: can you install llvm-15 here?

Comment (tokatoka) [reply .github/workflows/build_and_test.yml:124]: also or macos

**Comment by tokatoka** [comment]
it works now? 😉

**Comment by domenukk** [comment]
@tokatoka Thanks for figuring that out! <3

**Pull request merged**